tests/acceptance/TestHelpers/UploadHelper.php

```diff
@@ -225,58 +153,24 @@ public static function uploadWithAllMechanisms(
 		?string $destination,
 		?string $xRequestId = '',
 		?bool $overwriteMode = false,
-		?string $exceptChunkingType = ''
+		?int $davPathVersion = WebDavHelper::DAV_VERSION_NEW
 	):array {
 		$responses = [];
-		foreach ([1, 2] as $davPathVersion) {
-			if ($davPathVersion === 1) {
-				$davHuman = 'old';
-			} else {
-				$davHuman = 'new';
-			}
-
-			switch ($exceptChunkingType) {
-				case 'old':
-					$exceptChunkingVersion = 1;
-					break;
-				case 'new':
-					$exceptChunkingVersion = 2;
-					break;
-				default:
-					$exceptChunkingVersion = -1;
-					break;
-			}
-
-			foreach ([null, 1, 2] as $chunkingVersion) {
-				if ($chunkingVersion === $exceptChunkingVersion) {
-					continue;
-				}
-				$valid = WebDavHelper::isValidDavChunkingCombination(
-					$davPathVersion,
-					$chunkingVersion
-				);
-				if ($valid === false) {
-					continue;
-				}
-				$finalDestination = $destination;
-				if (!$overwriteMode && $chunkingVersion !== null) {
-					$finalDestination .= "-{$davHuman}dav-{$davHuman}chunking";
-				} elseif (!$overwriteMode && $chunkingVersion === null) {
-					$finalDestination .= "-{$davHuman}dav-regular";
-				}
-				$responses[] = self::upload(
-					$baseUrl,
-					$user,
-					$password,
-					$source,
-					$finalDestination,
-					$xRequestId,
-					[],
-					$davPathVersion,
-					$chunkingVersion,
-					2
-				);
-			}
+		foreach ([false, true] as $chunkingUse) {
+			$finalDestination = $overwriteMode ? $destination : $destination . ($chunkingUse ? "-chunked" : "");
```

Comment:
AFAIK, we should not need this.
```suggestion
```

Reply:
if the mentioned step fails, we have to look at what is happening and maybe scenario is not clear.
as per the scenario, I thought the the different upload methods to create same file will not creat file versions.

Reply:
let's have a look together

Reply:
After having closer look together with @saw-jan:
- refactored the scenarios from:
```gherkin
 Scenario Outline: upload file and no version is available using various chunking methods (except new chunking),
 Scenario Outline: upload a file twice and versions are available using various chunking methods (except new chunking)
```
to:
```gherkin
Scenario Outline: no version is available while uploading a file with multiple chunks
Scenario Outline: versions are available while uploading a file twice with multiple chunks
``` 
respectively.
- removed `uploadWithAllMechanisms` as it will be unused.
- changed return types to `ResponseInterface` for consistency and better error handling.